### PR TITLE
netlify-cli: 12.2.4 -> 17.37.1, migrate to buildNpmPackage

### PR DESCRIPTION
**Pull request by ShadowRZ**
https://github.com/netlify/cli/blob/v17.37.1/CHANGELOG.md

Turned out migrating to `buildNpmPackage` was much easier than I thought.

Closes #336530

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/web/netlify-cli/default.nix [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
+    tests.test = callPackage ./test.nix { };
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "Netlify command line tool";
+    homepage = "https://github.com/netlify/cli";
+    changelog = "https://github.com/netlify/cli/blob/v${version}/CHANGELOG.md";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ winter ];
```
Comment (SuperSandro2000) [inline pkgs/development/web/netlify-cli/default.nix:37]: Thats not you, or is it?

Comment (ShadowRZ) [reply pkgs/development/web/netlify-cli/default.nix:37]: Whoops, would fix shortly. 😅

**Comment by ShadowRZ** [comment]
Fixed meta infomations.

**Review by roberth — APPROVED** [review]
Amazing!
I've tried before, without success: https://github.com/NixOS/nixpkgs/pull/336530

Tested:
- :heavy_check_mark: `tests.test` attribute
  - :heavy_check_mark:  x86_64-linux
  - :x: aarch64-darwin
    - :heavy_check_mark: run `netlify dev` by hand
- :heavy_check_mark: push a preview deployment

I haven't managed to debug the darwin test failure, so a manual test will have to do, because this package is in dire need of an update.

**Comment by roberth** [comment]
Thank you @ShadowRZ!

**Pull request merged**